### ISSUE TITLE
Add "check-type-operator" option for whitespace rule

### ISSUE THREAD

**Issue by adidahiya** [palantir/tslint#488]
I'd like to lint for whitespace around type operators like `|` and `&`.
### Test Cases

passing:

``` ts
type StringOrNumber = string | number;

interface IFoo {
    someProp: string | number;
}

function foo(x: string | number) {
    //
}
```

failures:

``` ts
type StringOrNumber = string|number;

interface IFoo {
    someProp: string|number;
}

function foo(x: string|number) {
    //
}
```

relevant: TS repo has a custom rule for this already; we can just copy the implementation: https://github.com/Microsoft/TypeScript/pull/5366

**Comment by ajafff** [comment]
Fixed by #3083
